src/core/net/udp6.cpp

```diff
@@ -327,6 +327,15 @@ otError Udp::SendTo(SocketHandle &aSocket, Message &aMessage, const MessageInfo
     if (!IsMlePort(aSocket.mSockName.mPort) &&
         !(aSocket.mSockName.mPort == Tmf::kUdpPort && aMessage.GetSubType() == Message::kSubTypeJoinerEntrust))
     {
+        // Change anycast address to unicast address here because host may be unclear about the anycast address
```

Comment:
Just curious - in what cases can this occur?

Reply:
On-mesh commissioner sends petition to leader with ALOC, and since the TMF is running on host IP stack which doesn't know it is an ALOC, so it will respond with the ALOC as source address.

Reply:
Can/should we update TMF to not use ALOC as source address?

Reply:
I think so. But it seems preventing use anycast as source address is a general issue. To prevent this issue occurring again, I think it might be solved here. Thoughts?

Reply:
I'm wondering if it is better to return error here rather than trying to fix it automatically and "hide it under the covers". Thoughts?

Reply:
It's quite common for us to handle TMF and send response using the same `aMessageInfo`, in which the peer address might be an ALOC.

Reply:
SGTM - can we update the comment to something like: "Replace anycast address with a valid unicast address since response messages typically copy the peer address."?

Reply:
```suggestion
        // Replace anycast address with a valid unicast address since response messages typically copy the peer address
```